src/app/services/test-types/test-types.service.ts

```diff
@@ -32,4 +32,19 @@ export class TestTypesService extends TestTypesApiService {
   testTypeIdChanged(testTypeId: string): void {
     this.store.dispatch(testTypeIdChanged({ testTypeId }));
   }
+
+  findTestTypeNameById(id: string, testTypes: Array<TestType | TestTypeCategory>): TestType | undefined {
+    function idMatch(testType: TestType | TestTypeCategory) {
+      if (testType.id === id) {
+        result = testType;
+        return true;
+      }
+
+      return testType.hasOwnProperty('nextTestTypesOrCategories') && (testType as TestTypeCategory).nextTestTypesOrCategories!!.some(idMatch);
+    }
+
+    let result;
+    testTypes.some(idMatch);
+    return result;
+  }
```

Comment:
It is bad practice to update the value of an object by reference.
```suggestion
  findTestTypeNameById(id: string, testTypes: Array<TestType | TestTypeCategory>): TestType | undefined {
    function usingIdMatch(testType: TestType | TestTypeCategory) {
      return testType.id === id
        || testType.hasOwnProperty('nextTestTypesOrCategories')
        && (testType as TestTypeCategory).nextTestTypesOrCategories?.some(usingIdMatch);
    }

    return testTypes.find(usingIdMatch);
  }
```